site/app/controllers/HomePageController.php

```diff
@@ -290,10 +290,40 @@ public function createCoursePage() {
                 'showCourseCreationPage',
                 $faculty ?? null,
                 $this->core->getUser()->getId(),
-                $this->core->getQueries()->getAllUnarchivedSemester(),
+                //$this->core->getQueries()->getAllUnarchivedSemester(),
+                $this->core->getQueries()->getAllTerms(),
                 $this->core->getUser()->getAccessLevel() === User::LEVEL_SUPERUSER,
                 $this->core->getCsrfToken()
             )
         );
     }
+
+    /**
+     * @Route("/term/new", methods={"POST"})
+     * @return Response
+     */
+    public function addNewTerm() {
+        if (isset($_POST['term_id']) && isset($_POST['term_name']) && isset($_POST['start_date']) && isset($_POST['end_date'])) {
+            $term_id = $_POST['term_id'];
+            $term_name = $_POST['term_name'];
+            $start_date = $_POST['start_date'];
+            $end_date = $_POST['end_date'];
+
+            $terms = $this->core->getQueries()->getAllTerms();
+            if (in_array($term_id, $terms)) {
+                $this->core->addErrorMessage("Term id already exists.");
+            }
+            else if ($end_date < $start_date) {
+                $this->core->addErrorMessage("End date should be after Start date.");
+            }
+            else {
+                $this->core->getQueries()->createNewTerm($term_id, $term_name, $start_date, $end_date);
+                $this->core->addSuccessMessage("Term added successfully.");
+            }
+
+            return Response::RedirectOnlyResponse(
```

Comment:
I would move this outside of the if statement on line 306 and then add an else with an error message to that if statement as well.

Reply:
this if statement will always evaluate to true since all inputs will be set as they're all required, so i guess there is no need for an else statement